### PR TITLE
[Feature] Enable PTPC FP8 for compressed tensors moe (aiter kernel)

### PR DESCRIPTION
**Pull request by qichu-yun**
## Motivation

Deepseek-R1 PTPC FP8 model for llm-compressor quantized model can be supported on MI308.

## Accuracy Tests
Here is my script.
```shell
model=/data/models/Deepseek-R1-FP8-Dynamic
TP=8
EP=1

python3 -m sglang.launch_server \
    --model-path ${model} \
    --host localhost \
    --port 9000 \
    --tp-size ${TP} \
    --ep-size ${EP} \
    --trust-remote-code \
    --chunked-prefill-size 16384 \
    --mem-fraction-static 0.85 \
    --disable-radix-cache \
    --max-prefill-tokens 16384 \
    --cuda-graph-max-bs 128 \
    2>&1 | tee log.server.log &
``` 
Parallel strategy：TP8
acc:
```shell
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.932|±  |0.0160|
|     |       |strict-match    |     5|exact_match|↑  |0.928|±  |0.0164|
``` 

Parallel strategy：TP8 + EP8
acc:
```shell
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.967|±  |0.0112|
|     |       |strict-match    |     5|exact_match|↑  |0.967|±  |0.0112|
``` 

## Benchmarking
Here is my script.
```
export SGLANG_USE_AITER=1
model=/mnt/raid0/models/deepseek-r1-FP8-Dynamic-from-BF16

input_tokens=3584
output_tokens=1024
max_concurrency=64
num_prompts=128

echo "bench model: ${model}"
echo "input tokens: ${input_tokens}"
echo "output tokens: ${output_tokens}"
echo "max concurrency: ${max_concurrency}"
echo "num prompts: ${num_prompts}"

python3 -m sglang.bench_serving \
    --host localhost \
    --port 9000 \
    --model ${model} \
    --dataset-name random \
    --random-input ${input_tokens} \
    --random-output ${output_tokens} \
    --random-range-ratio 1.0 \
    --max-concurrency ${max_concurrency} \
    --num-prompt ${num_prompts} \
```

Parallel strategy：TP8

```shell
============ Serving Benchmark Result ============
Backend:                                 sglang    
Traffic request rate:                    inf       
Max request concurrency:                 64        
Successful requests:                     128       
Benchmark duration (s):                  152.51    
Total input tokens:                      458752    
Total input text tokens:                 458752    
Total input vision tokens:               0         
Total generated tokens:                  131072    
Total generated tokens (retokenized):    130687    
Request throughput (req/s):              0.84      
Input token throughput (tok/s):          3007.92   
Output token throughput (tok/s):         859.41    
Total token throughput (tok/s):          3867.33   
Concurrency:                             63.97     
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   76217.67  
Median E2E Latency (ms):                 76215.98  
---------------Time to First Token----------------
Mean TTFT (ms):                          11564.82  
Median TTFT (ms):                        11734.60  
P99 TTFT (ms):                           20507.28  
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          63.20     
Median TPOT (ms):                        63.04     
P99 TPOT (ms):                           73.21     
---------------Inter-Token Latency----------------
Mean ITL (ms):                           63.20     
Median ITL (ms):                         54.44     
P95 ITL (ms):                            55.93     
P99 ITL (ms):                            56.77     
Max ITL (ms):                            19669.39  
==================================================
``` 

Parallel strategy：TP8 + EP8

```shell
============ Serving Benchmark Result ============
Backend:                                 sglang    
Traffic request rate:                    inf       
Max request concurrency:                 64        
Successful requests:                     128       
Benchmark duration (s):                  155.88    
Total input tokens:                      458752    
Total input text tokens:                 458752    
Total input vision tokens:               0         
Total generated tokens:                  131072    
Total generated tokens (retokenized):    64131     
Request throughput (req/s):              0.82      
Input token throughput (tok/s):          2942.89   
Output token throughput (tok/s):         840.83    
Total token throughput (tok/s):          3783.72   
Concurrency:                             63.96     
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   77898.83  
Median E2E Latency (ms):                 77897.51  
---------------Time to First Token----------------
Mean TTFT (ms):                          10522.66  
Median TTFT (ms):                        10886.52  
P99 TTFT (ms):                           18062.87  
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          65.86     
Median TPOT (ms):                        65.48     
P99 TPOT (ms):                           75.02     
---------------Inter-Token Latency----------------
Mean ITL (ms):                           65.86     
Median ITL (ms):                         58.54     
P95 ITL (ms):                            60.12     
P99 ITL (ms):                            61.09     
Max ITL (ms):                            17324.30  
==================================================

``` 
As you can see, TP8 has better performance than TP8+EP8, and we recommend running TP8 for the PTPC model.

### CODE REVIEW

**Comment by qichu-yun** [comment]
Could you please review this PR when you're free, thanks! ? @wuhuikx @zejunchen-zejun @HaiShaw

File: python/sglang/srt/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py [hunk 1]
```diff
@@ -101,6 +104,17 @@ def process_weights_after_loading(self, layer) -> None:
...
+            from aiter.ops.shuffle import shuffle_weight
+
+            # keep the weight as (N, K)
+            layer.weight = Parameter(
+                shuffle_weight(weight, layout=(16, 16)), requires_grad=False
+            )
+        else:
+            # keep the weight as (K, N)
+            layer.weight = Parameter(weight.t(), requires_grad=False)
+
```
[the remark this answers is not in the document]
Comment (HaiShaw) [reply python/sglang/srt/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py]: @qichu-yun please double check this. Thanks!

**Review by HaiShaw — CHANGES_REQUESTED** [review]
Please address comment left

File: python/sglang/srt/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (qichu-yun) [reply python/sglang/srt/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py]: I have remove the .t() with the help of Bowen.

File: python/sglang/srt/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py [hunk 2]
```diff
@@ -101,6 +102,15 @@ def process_weights_after_loading(self, layer) -> None:
...
 
+        if _use_aiter:
+            # keep the weight as (N, K)
+            layer.weight = Parameter(
+                shuffle_weight(weight, layout=(16, 16)), requires_grad=False
+            )
+        else:
+            # keep the weight as (K, N)
+            layer.weight = Parameter(weight.t(), requires_grad=False)
+
```
Comment (HaiShaw) [inline python/sglang/srt/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py]: L105-L113 is redundant to L86-L91 in one case, if so (not intended to do twice), then need to refactor code.

File: python/sglang/srt/layers/quantization/compressed_tensors/compressed_tensors_moe.py [hunk 3]
```diff
@@ -277,16 +284,77 @@ def process_weights_after_loading(self, layer: torch.nn.Module | FusedMoE) -> No
                 max_w13_scales, requires_grad=False
             )
 
-        if self.weight_quant.strategy == QuantizationStrategy.CHANNEL and _use_aiter:
+        if _use_aiter:
+            padding_size = get_int_env_var("AITER_MOE_PADDING_SIZE")
```
Comment (HaiShaw) [inline python/sglang/srt/layers/quantization/compressed_tensors/compressed_tensors_moe.py]: `AITER_MOE_PADDING_SIZE` - undocumented ENV introduced, not have its description, neither tests in PR body.
Please document it, describe its purpose.

Note - see if this serves the purpose:
https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/layers/moe/fused_moe_triton/fused_moe.py#L58

**Review by HaiShaw — CHANGES_REQUESTED** [review]

File: python/sglang/srt/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py (hunk 2 above)
Comment (qichu-yun) [reply python/sglang/srt/layers/quantization/compressed_tensors/schemes/compressed_tensors_w8a8_fp8.py]: I have organized the code uniformly under the “elif self.strategy == QuantizationStrategy.CHANNEL:” branch.

File: python/sglang/srt/layers/quantization/compressed_tensors/compressed_tensors_moe.py (hunk 3 above)
Comment (qichu-yun) [reply python/sglang/srt/layers/quantization/compressed_tensors/compressed_tensors_moe.py]: I have removed the env AITER_MOE_PADDING_SIZE. There is no need to use this env in this case.

**Comment by HaiShaw** [comment]
@qichu-yun why keep the `torch.nn.functional.pad` calls if nothing really to be padded?

**Review by HaiShaw — APPROVED** [review]
